src/ifirmware.h

```diff
@@ -3,6 +3,14 @@
 #include <QObject>
 #include <QString>
 
+struct PrinterStatus {
```

Comment:
because this structure is useful outside of iFirmware. If not as a struct perhaps we should make it a type.